### PR TITLE
fix: skip simulation metrics if simulation disabled or chain not supported

### PR DESCRIPTION
**Pull request by matthewwalsh0**
## **Description**

Skip simulation metrics if simulation is disabled or the transaction chain does not support simulation.

Also adopt breaking changes from the `TransactionController` that replace the `isReverted` property with suitable error codes provided by the new `SimulationErrorCode` enum.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/23713?quickstart=1)

## **Related issues**

Fixes: [#2297](https://github.com/MetaMask/MetaMask-planning/issues/2297)

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: package.json
Comment (dbrans) [inline package.json:235]: Remove before merging?

File: ui/pages/confirmations/components/simulation-details/simulation-details.tsx
Comment (dbrans) [inline ui/pages/confirmations/components/simulation-details/simulation-details.tsx:61]:
```suggestion
  const message = error.code === SimulationErrorCode.Reverted
      ? t('simulationDetailsTransactionReverted')
      : t('simulationDetailsFailed');
```

File: ui/pages/confirmations/components/simulation-details/useSimulationMetrics.ts
Comment (dbrans) [inline ui/pages/confirmations/components/simulation-details/useSimulationMetrics.ts:85]: Should we make this array an exported const somewhere?

File: ui/pages/confirmations/components/simulation-details/simulation-details.tsx
Comment (dbrans) [inline ui/pages/confirmations/components/simulation-details/simulation-details.tsx:201]: I think it might make sense to move all the error handling inside the error component:
https://github.com/MetaMask/metamask-extension/pull/23726/files

File: package.json
Comment (matthewwalsh0) [reply package.json]: Done, but this is blocked until Goktug's PR goes in which handles the `GasFeeController` breaking changes.

Unfortunately this likely means we'll either need those changes in the release branch, or a transaction controller patch.

File: ui/pages/confirmations/components/simulation-details/simulation-details.tsx
Comment (matthewwalsh0) [reply ui/pages/confirmations/components/simulation-details/simulation-details.tsx:61]: Not sure what you mean here?

This previously returned so I'm doing the same but with a ternary.

File: ui/pages/confirmations/components/simulation-details/useSimulationMetrics.ts
Comment (matthewwalsh0) [reply ui/pages/confirmations/components/simulation-details/useSimulationMetrics.ts:85]: I considered it, but this and the list in `SimulationDetails` for hiding are ultimately distinct checks as one is saying what errors should hide the component, where as this is what error states result in metrics we don't want.

Hence why I thought it would be more explicit to do those checks inline without a constant.

File: ui/pages/confirmations/components/simulation-details/simulation-details.tsx
Comment (matthewwalsh0) [reply ui/pages/confirmations/components/simulation-details/simulation-details.tsx:201]: For the sake of the bug fix, ideally we'd do the minimum here hence why I've just changed the condition to rely on the error code as that could also be considered a bug.

If we move the rendering around in future, I'd still lean towards keeping this condition at the top level as it results in nothing being rendered, where as a component function implies something would be rendered within the component.

File: ui/pages/confirmations/components/simulation-details/simulation-details.tsx
Comment (dbrans) [reply ui/pages/confirmations/components/simulation-details/simulation-details.tsx:61]: Hey, my bad for not providing more context. Just a small nit - with the ternary operator, we just have a single expression and can replace the getMessage function with `const message = ...`. It's very minor.

File: ui/pages/confirmations/components/simulation-details/simulation-details.tsx
Comment (dbrans) [reply ui/pages/confirmations/components/simulation-details/simulation-details.tsx:201]: > I'd still lean towards keeping this condition at the top level as it results in nothing being rendered

That's a very good point. Thanks for walking me through your thought process on that.

**Review by dbrans — DISMISSED** [review]

**Comment by matthewwalsh0** [comment]
@metamaskbot update-policies

**Review by OGPoyraz — DISMISSED** [review]

**Review by OGPoyraz — APPROVED** [review]

File: ui/pages/confirmations/components/simulation-details/simulation-details.tsx
Comment (dbrans) [inline ui/pages/confirmations/components/simulation-details/simulation-details.tsx:178]: Nice

File: ui/pages/confirmations/components/simulation-details/useSimulationMetrics.ts
Comment (dbrans) [inline ui/pages/confirmations/components/simulation-details/useSimulationMetrics.ts:131]: Didn't notice `JSON.stringify(params)`  before. Nice.

**Review by dbrans — APPROVED** [review]

**Pull request merged**
